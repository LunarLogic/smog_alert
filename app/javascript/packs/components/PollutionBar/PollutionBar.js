import React from "react";
import { PropTypes } from "prop-types";

import "./PollutionBar.scss";

const PollutionBar = ({ width, backgroundColor, location, value }) => {
  return (
    <div className="pollution-bar">
      <div className="pollution-bar__info">
        <div className="pollution-bar__info-location">{location}</div>
        <div className="pollution-bar__info-type">PM10</div>
        <div className="pollution-bar__info-value">{value} μg/m³</div>
      </div>
      <div
        className="pollution-bar__bar"
        style={{ width: `${width}%`, backgroundColor: `${backgroundColor}` }}
      ></div>
    </div>
  );
};

PollutionBar.propTypes = {
  width: PropTypes.number,
  backgroundColor: PropTypes.string,
  location: PropTypes.string,
  value: PropTypes.number
};

export default PollutionBar;
