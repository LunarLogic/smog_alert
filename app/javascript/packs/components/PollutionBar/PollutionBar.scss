@import "../../styles/variables";
.pollution-bar {
  height: 8.3rem;
  &__info {
    font-size: 1.6rem;
    line-height: 1.9rem;
    font-weight: 500;

    display: flex;
    &-location {
      margin-right: 2.2rem;
    }
  }
  &__bar {
    height: 2.2rem;
    border-radius: 1.1rem;
    margin-top: 0.9rem;
  }
  > hr {
    margin: 1.7rem 0;
    border: 0.05rem solid $frameColor;
  }
}
