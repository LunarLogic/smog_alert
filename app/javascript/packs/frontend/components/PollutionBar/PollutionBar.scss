@import "../../styles/variables";

.pollution-bar {
  padding: 1.95rem 2.4rem;
  border-bottom: 1px solid $frameColor;
  cursor: pointer;

  &:hover {
    background-color: $greyLighter;

    .pollution-bar__info-location {
      color: $secondaryColor;
    }
  }
  &__info {
    display: flex;
    &-location {
      font-size: 1.8rem;
      line-height: 2.1rem;
      font-weight: bold;
      color: $textColor;
      margin-right: 2.2rem;
    }
    &-type,
    &-value {
      font-size: 1.6rem;
      line-height: 2.1rem;
      color: $grey;
    }
    &-type {
      font-weight: bold;
      margin-right: 0.7rem;
    }
  }
  &__bar {
    height: 1.8rem;
    border-radius: 0.9rem;
    margin-top: 0.9rem;
  }
}
