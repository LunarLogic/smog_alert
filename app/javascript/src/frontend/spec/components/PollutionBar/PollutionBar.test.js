import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

import PollutionBar from "../../../components/PollutionBar/PollutionBar";

describe("<PollutionBar />", () => {
  it("displays given props data", () => {
    const wrapper = mount(
      <PollutionBar
        location="Brzezie"
        value={25}
        width={30}
        backgroundColor="#7d0d0f"
      />
    );
    expect(wrapper.find("div.pollution-bar__info-location").text()).toBe(
      "Brzezie"
    );
    expect(wrapper.find("div.pollution-bar__info-value").text()).toBe(
      "25 μg/m³"
    );
    expect(wrapper.find("div.pollution-bar__bar").prop("style")).toHaveProperty(
      "width",
      "30%"
    );
    expect(wrapper.find("div.pollution-bar__bar").prop("style")).toHaveProperty(
      "backgroundColor",
      "#7d0d0f"
    );
  });
  it("expect to render PollutionBar component", () => {
    const wrapper = shallow(
      <PollutionBar
        location="Brzezie"
        value={25}
        width={30}
        backgroundColor="#7d0d0f"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
