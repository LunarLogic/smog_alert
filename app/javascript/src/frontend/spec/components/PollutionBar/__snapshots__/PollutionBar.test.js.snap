// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PollutionBar /> expect to render PollutionBar component 1`] = `
<div
  className="pollution-bar"
>
  <div
    className="pollution-bar__info"
  >
    <div
      className="pollution-bar__info-location"
    >
      Brzezie
    </div>
    <div
      className="pollution-bar__info-type"
    >
      PM10
    </div>
    <div
      className="pollution-bar__info-value"
    >
      25
       μg/m³
    </div>
  </div>
  <div
    className="pollution-bar__bar"
    style={
      Object {
        "backgroundColor": "#7d0d0f",
        "width": "30%",
      }
    }
  />
</div>
`;
